test/Renci.SshNet.IntegrationTests/CompressionTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Renci.SshNet.Compression;
+
+namespace Renci.SshNet.IntegrationTests
+{
+    [TestClass]
+    public class CompressionTests : IntegrationTestBase
+    {
+        private IConnectionInfoFactory _connectionInfoFactory;
+
+        [TestInitialize]
+        public void SetUp()
+        {
+            _connectionInfoFactory = new LinuxVMConnectionFactory(SshServerHostName, SshServerPort);
+        }
+
+        [TestMethod]
+        public void None()
+        {
+            DoTest(new KeyValuePair<string, Func<Compressor>>("none", null));
+        }
+
+#if NET6_0_OR_GREATER
+        [TestMethod]
+        public void ZlibOpenSsh()
+        {
+            DoTest(new KeyValuePair<string, Func<Compressor>>("zlib@openssh.com", () => new ZlibOpenSsh()));
+        }
+#endif
+
+        private void DoTest(KeyValuePair<string, Func<Compressor>> compressor)
+        {
+            using (var scpClient = new ScpClient(_connectionInfoFactory.Create()))
+            {
+                scpClient.ConnectionInfo.CompressionAlgorithms.Clear();
+                scpClient.ConnectionInfo.CompressionAlgorithms.Add(compressor);
```

Comment:
nit: in e.g. CipherTests, HmacTests, we specify the algorithm on the server and test the default SSH.NET configuration against it, rather than specifying the algorithm on the client.

Reply:
The server can only be configured to compression true/false